### PR TITLE
Added documentation for core.args and stdlib.log

### PR DESCRIPTION
**Pull request by Arignir**

### CODE REVIEW

File: core/args.py [hunk 1]
```diff
@@ -13,9 +11,9 @@
 
 
 def parse_args():
-    """
-    Parses the command line arguments.
-    This function may exit if the command line arguments are invalid.
+    """Parse the command line arguments.
```
Comment (doom) [inline core/args.py:14]: You seem to have switched from "Parses" to "Parse", is that intentional ? If so, you should also update the other comments, in order to keep a consistent commenting style.

File: core/__init__.py [hunk 2]
```diff
@@ -1,7 +1,10 @@
 #!/usr/bin/env python3.6
 # -*- coding: utf-8 -*-
 
+"""Core modules that are used exclusively by :py:mod:`nbuild` or the compilation library (:py:mod:`stdlib`).
+
+The content of the core module shouldn't be used by a build manifest.
```
Comment (doom) [inline core/__init__.py]: I think "should not" would be more appropriate here (this is also valid for other occurrences of "shouldn't" in this PR).

File: core/__init__.py [hunk 3]
```diff
@@ -1,7 +1,10 @@
 #!/usr/bin/env python3.6
 # -*- coding: utf-8 -*-
 
+"""Core modules that are used exclusively by :py:mod:`nbuild` or the compilation library (:py:mod:`stdlib`).
```
Comment (doom) [inline core/__init__.py]: "Core modules used exclusively ..." sounds better

File: core/cache.py [hunk 4]
```diff
@@ -12,10 +12,12 @@
...
 
 
-def get_install_cache(build: stdlib.Build):
-    """
-    Returns the location of the cache where the files produced by the given build should be placed.
-    The content of this cache mimics a Linux file system.
+def get_install_cache(build: stdlib.Build) -> str:
+    """Returns a path pointing to the cache where the files produced by the given build should be stored.
+
+    :info: The content of this cache mimics a Linux file system.
```
Comment (doom) [inline core/cache.py]: Here you put a `.` at the end of the line...

File: core/cache.py [hunk 5]
```diff
@@ -12,10 +12,12 @@
...
+    :param build: The build associated with the cache
```
Comment (doom) [inline core/cache.py:19]: But here you don't, which one is correct ?

File: docs/Makefile [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+# Put it first so that "make" without argument is like "make help".
+help:
+	@$(SPHINXBUILD) -M help "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS) $(O)
+
+.PHONY: help Makefile
+
+# Catch-all target: route all unknown targets to Sphinx using the new
+# "make mode" option.  $(O) is meant as a shortcut for $(SPHINXOPTS).
+%: Makefile
+	@$(SPHINXBUILD) -M $@ "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS) $(O)
```
Comment (doom) [inline docs/Makefile:20]: I like newlines at the end of my files, don't you ? :p

File: docs/source/index.rst [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+   sphinx-quickstart on Sun Feb 10 00:55:52 2019.
+   You can adapt this file completely to your liking, but it should at least
+   contain the root `toctree` directive.
+
+Welcome to NBuild's documentation!
+==================================
+
+This documentation covers all types and functions that can be found in a build manifest or in nbuild itself.
+
+It is made of two major modules: the :py:mod:`core` module and the :py:mod:`stdlib` module. The first one contains functions only
```
Comment (doom) [inline docs/source/index.rst]: Why break the line here ?

File: docs/source/index.rst [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+used by :py:mod:`nbuild` or by :py:mod:`stdlib` and shouldn't be used by a build manifest. On the over hand, :py:mod:`stdlib` is made to be used by build manifests.
```
Comment (doom) [inline docs/source/index.rst]: "on the other hand"

**Review by doom — CHANGES_REQUESTED** [review]

File: core/args.py (hunk 1 above)
Comment (Arignir) [reply core/args.py:14]: Yeah I try to have the first verb in the infinitive, but it's so unnatural to me that I fail sometimes. Thanks for spotting that.

File: docs/Makefile (hunk 6 above)
Comment (Arignir) [reply docs/Makefile:20]: (It's actually an automatically-generated Makefile :o)

**Review by doom — DISMISSED** [review]

File: README.md [hunk 9]
```diff
@@ -1 +1,28 @@
...
+
+Nest build requires python3.6+. To install its dependencies, run
+
+```bash
+$ pip install -r requirements.txt
+```
+
+## How it works
+
+Nest build uses a python file (called a build manifest) to describe how the compilation should be performed. The build manifest downloads the source code of a software, compiles it and splits it multiple packages.
```
Comment (Pedrochu) [inline README.md]: Should be `splis it into multiple packages` if that's what you meant

**Review by Pedrochu — CHANGES_REQUESTED** [review]

**Pull request merged**